api/app/GraphQL/Mutations/DownloadUserDoc.php

```diff
@@ -18,8 +19,21 @@ public function __invoke(null $_, array $args)
         $user = Auth::user();
         throw_unless(is_string($user?->id), UnauthorizedException::class);
 
+        $targetUser = User::find($args['id']);
+        $firstName = $targetUser?->first_name;
+        $lastName = $targetUser?->last_name;
+        if (isset($firstName)) {
+            $firstName = trim(filter_var($firstName, FILTER_SANITIZE_EMAIL));
+        }
+        if (isset($lastName)) {
+            $lastName = trim(filter_var($lastName, FILTER_SANITIZE_EMAIL));
+        }
+
         try {
-            $fileName = sprintf('%s_%s.docx', trim(__('filename.user')), date('Y-m-d_His'));
+
+            $fileName = $args['anonymous'] ?
+                sprintf('%s - %s - Profile - Profil.docx', $firstName ? $firstName : '', $lastName ? substr($lastName, 0, 1) : '') :
+                sprintf('%s - %s - Profile - Profil.docx', $firstName ? $firstName : '', $lastName ? $lastName : '');
```

Comment:
I have a minor concern that may not be an issue. But, we no longer have any unique identifier so if an admin downloads a file twice within the window for pruning, it may overwrite the previously file if there were changes.

Do we care about this or is it expected that we will just maintain the most recent file in that window?

Reply:
I have no idea